lua/mason-registry/hadolint/init.lua

```diff
@@ -0,0 +1,33 @@
+local Pkg = require "mason-core.package"
+local _ = require "mason-core.functional"
+local platform = require "mason-core.platform"
+local github = require "mason-core.managers.github"
+local std = require "mason-core.managers.std"
+
+local coalesce, when = _.coalesce, _.when
+
+return Pkg.new {
+    name = "hadolint",
+    desc = [[Dockerfile linter, validate inline bash, written in Haskell]],
+    homepage = "https://github.com/hadolint/hadolint",
+    languages = { Pkg.Lang.Dockerfile },
+    categories = { Pkg.Cat.Linter },
+    ---@async
+    ---@param ctx InstallContext
+    install = function(ctx)
+        github
+            .download_release_file({
+                repo = "hadolint/hadolint",
+                out_file = platform.is.win and "hadolint.exe" or "hadolint",
+                asset_file = coalesce(
+                    when(platform.is.mac_x64, "hadolint-Darwin-x86_64"),
```

Comment:
Mac x64 binaries can often be executed on the ARM chips as well! It seems like this one can, so let's do:

```suggestion
                    when(platform.is.mac, "hadolint-Darwin-x86_64"),
```